tests/risks/test_effect.py

```diff
@@ -279,9 +279,9 @@ def test_CategoricalRiskComponent_dichotomous_case(base_config, base_plugins):
 
     simulation.data.write("risk_factor.test_risk.relative_risk", rr_data)
     simulation.data.write("risk_factor.test_risk.population_attributable_fraction", 1)
-
     simulation.data.write("risk_factor.test_risk.affected_causes", affected_causes)
     simulation.data.write("risk_factor.test_risk.distribution", "dichotomous")
+    simulation.data.write("risk_factor.test_risk.affected_risk_factors", [])
 
```

Comment:
yep, I think we only required only `affectd_causes` for risk and `affected_causes` and `affected_risk_factors` for coverage_gap before James rewrote the risk_effect. Then it changes to require both from risk also. good catch.